### ISSUE TITLE
"preparing" as task state or sub-type of "waiting"?

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#4458]
(Putting this up for the record, although from in-meeting discussion we might leave it as-is ... in case the question gets raised again at some point in the future!).

In Cylc 7 and earlier, the `waiting` task state meant waiting on:
- other-task prerequisites
- clock or external triggers

Any other kind of "waiting" had a dedicated task state:
- `retrying`
- `queued`
- `held`
- `ready` (preparing job)
- `runahead`

In Cylc 8 we have (for good reasons) reclassified the `waiting` task state as waiting on almost every kind of "dependency":
- retry (via clock trigger)
- clock and external triggers
- queues
- task hold
- runahead limited

The odd one out is the `preparing` task state (formerly `ready`), which covers job script preparation and job submission via the scheduler process pool. To users, this is essentially an internal queue for job submission. Normally it is highly transient, but it can last a while if the process pool is congested.  

**QUESTION:** should `preparing` just be another sub-type of the `waiting` task state?

PROs?
- simplicity: `waiting` would mean waiting on absolutely anything that prevents job submission. And users can easily discover exactly what is being waited on
- consistency: is there really any solid reason to single out `preparing` alone, of all the "waiting" conditions, as a special task state? 

CONs?
- I can't really think of any cons, but at least the status quo isn't too hard to rationalize:
  - `preparing` is somewhat different to the other wait conditions, in that it is entirely internal (however, why does that justify a separate task state?)
  - `waiting` -> `preparing` -> `submitted`  is still a lot simpler than the Cylc 7 way
  -
